sklearn/linear_model/base.py

```diff
@@ -56,12 +57,19 @@ def make_dataset(X, y, sample_weight, random_state=None):
     # seed should never be 0 in SequentialDataset
     seed = rng.randint(1, np.iinfo(np.int32).max)
 
+    if isinstance(X, np.float32):
```

Comment:
Shouldn't you be checking the dtype, rather than X?

Something like X.dtype == np.float32